### PR TITLE
Add Openal-soft

### PR DESCRIPTION
**Pull request by martinRenou**
Signed-off-by: martinRenou <martin.renou@gmail.com>

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by martinRenou** [comment]
@conda-forge/help-c-cpp I guess this is ready for a review/merge :)

File: recipes/openal-soft/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  sha256: c32d10473457a8b545aab50070fe84be2b5b041e1f2099012777ee6be0057c13
+
+build:
+  number: 0
+  skip: True  # [win and vc<14]
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - cmake
```
Comment (chrisburr) [inline recipes/openal-soft/meta.yaml]:
```suggestion
    - cmake
    - make  # [unix]
```

File: recipes/openal-soft/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+test:
+  commands:
+    - test -f ${PREFIX}/lib/cmake/OpenAL/OpenALConfig.cmake                             # [unix]
+    - if exist %LIBRARY_LIB%\\cmake\\OpenAL\\OpenALConfig.cmake (exit 0) else (exit 1)  # [win]
+
+about:
+  home: https://github.com/kcat/openal-soft
+  license: LGPL
```
Comment (chrisburr) [inline recipes/openal-soft/meta.yaml]:
```suggestion
  license: LGPL-2.0-or-later
```

File: recipes/openal-soft/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - cmake
+
+test:
+  commands:
+    - test -f ${PREFIX}/lib/cmake/OpenAL/OpenALConfig.cmake                             # [unix]
+    - if exist %LIBRARY_LIB%\\cmake\\OpenAL\\OpenALConfig.cmake (exit 0) else (exit 1)  # [win]
```
Comment (chrisburr) [inline recipes/openal-soft/meta.yaml]: Can you add any tests for the commands that are? Generally people use `--help` or `--version` like
```suggestion
    - if exist %LIBRARY_LIB%\\cmake\\OpenAL\\OpenALConfig.cmake (exit 0) else (exit 1)  # [win]
    - altonegen --version
    - alrecord --help
    - openal-info
```

**Comment by martinRenou** [comment]
I commited all your suggestions, thanks!

**Comment by martinRenou** [comment]
Removing the `altogen --version` test as it fails on the CI (no device available)

**Comment by martinRenou** [comment]
Even `altonegen --help` fails with "no device available", let's just test for the binary existence

**Comment by martinRenou** [comment]
Green!

**Comment by martinRenou** [comment]
Thanks a lot for your help @chrisburr!

**Pull request merged**